crypto/dh/dh_lib.c

```diff
@@ -168,6 +168,11 @@ int DH_up_ref(DH *r)
     return ((i > 1) ? 1 : 0);
 }
 
+void ossl_dh_set0_libctx(DH *d, OSSL_LIB_CTX *libctx)
```

Comment:
I'm wondering if these should also take a propq argument?  More for consistency than necessity.  I can't think why it would be needed but not having it if it is is bad.

Reply:
Another PR, then, and should be separate function (a set1, since we most probably will strdup it).

Reply:
I do not really see why we would need it. propq is attribute of an operation (or fetch for an operation) not of a key object. The pairing of libctx and propq is just a coincidence and not a real thing. We already discussed this a few months ago.